### PR TITLE
Fix SAML decoding issue

### PR DESCRIPTION
**Pull request by leleuj**

### CODE REVIEW

File: support/cas-server-support-saml/src/test/java/org/apereo/cas/support/saml/util/Saml20ObjectBuilderTests.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests the {@link AbstractSaml20ObjectBuilder}.
+ *
+ * @author Jerome Leleu
+ * @since 5.2.9
+ */
+public class Saml20ObjectBuilderTests {
```
Comment (mmoayyed) [inline support/cas-server-support-saml/src/test/java/org/apereo/cas/support/saml/util/Saml20ObjectBuilderTests.java]: I suspect this should be added to the test suite for the project.

File: support/cas-server-support-saml/src/main/java/org/apereo/cas/support/saml/util/AbstractSaml20ObjectBuilder.java [hunk 2]
```diff
@@ -442,8 +442,8 @@ public String decodeSamlAuthnRequest(final String encodedRequestXmlString) {
         final String inflated = CompressionUtils.inflate(decodedBytes);
         if (!StringUtils.isEmpty(inflated)) {
             return inflated;
+        } else {
```
Comment (mmoayyed) [inline support/cas-server-support-saml/src/main/java/org/apereo/cas/support/saml/util/AbstractSaml20ObjectBuilder.java]: This is fine, but the else clause is redundant here.

File: support/cas-server-support-saml/src/test/java/org/apereo/cas/support/saml/util/Saml20ObjectBuilderTests.java (hunk 1 above)
Comment (leleuj) [reply support/cas-server-support-saml/src/test/java/org/apereo/cas/support/saml/util/Saml20ObjectBuilderTests.java]: Yes. Will do.

File: support/cas-server-support-saml/src/main/java/org/apereo/cas/support/saml/util/AbstractSaml20ObjectBuilder.java (hunk 2 above)
Comment (leleuj) [reply support/cas-server-support-saml/src/main/java/org/apereo/cas/support/saml/util/AbstractSaml20ObjectBuilder.java]: A matter of style I would say, but yes, let's remove the `else` clause...

**Comment by leleuj** [comment]
Port forward: https://github.com/apereo/cas/pull/3595

@mmoayyed I accounted for your comments. I also removed the non-working `inflate`method to use the `decodeByteArrayToString` method instead (renamed as `inflate` as it was really an inflation operation).

**Comment by mmoayyed** [comment]
The changes introduced here break GoogleApps functionality:
https://groups.google.com/a/apereo.org/forum/?utm_medium=email&utm_source=footer#!msg/cas-user/wNghzsq6QFs/VjH6bsNJCQAJ

Reverting.

**Pull request merged**